### PR TITLE
[ATMOSPHERE-589] Add support for nfs_shares_config

### PR DESCRIPTION
**Pull request by larainema**

### CODE REVIEW

**Review by mnaser — CHANGES_REQUESTED** [review]
One last thing missing, we actually need to _replace_ the value of `nfs_shares_config` in the final generated config file to become `/etc/cinder/nfs/{{ $key }}` so that it points to that mount, cause right now it's actually just going to the data still.

File: charts/cinder/templates/configmap-etc.yaml [hunk 1]
```diff
@@ -198,3 +198,9 @@ data:
   external-ceph.conf: {{ include "helm-toolkit.utils.to_oslo_conf" .Values.ceph_client.external_ceph.conf | b64enc }}
 {{- end }}
 {{- end }}
+{{- range $key, $backend := .Values.conf.backends }}
+{{- if $backend.nfs_shares_config }}
+  nfs_shares_{{ $key }}: |
```
[the remark this answers is not in the document]
Comment (larainema) [reply charts/cinder/templates/configmap-etc.yaml:203]: I think configmap data key should consistent with mount subPath, right?

File: charts/cinder/templates/deployment-volume.yaml [hunk 2]
```diff
@@ -242,6 +242,14 @@ spec:
               readOnly: true
             {{- end }}
             {{- end }}
+            {{- range $key, $backend := .Values.conf.backends }}
+            {{- if $backend.nfs_shares_config }}
+            - name: cinder-etc
+              mountPath: /etc/cinder/nfs/nfs_shares_{{ $key }}
```
[the remark this answers is not in the document]
Comment (larainema) [reply charts/cinder/templates/deployment-volume.yaml]: done

File: charts/cinder/templates/configmap-etc.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (larainema) [reply charts/cinder/templates/configmap-etc.yaml]: done

File: charts/patches/cinder/0002-add-nfs-support.patch [hunk 3]
```diff
@@ -0,0 +1,52 @@
+diff --git a/charts/cinder/templates/configmap-etc.yaml b/charts/cinder/templates/configmap-etc.yaml
+index 1a20ea84..97bfe0f7 100644
+--- a/charts/cinder/templates/configmap-etc.yaml
++++ b/charts/cinder/templates/configmap-etc.yaml
```
Comment (okozachenko1203) [inline charts/patches/cinder/0002-add-nfs-support.patch:4]:
```suggestion
diff --git a/cinder/templates/configmap-etc.yaml b/cinder/templates/configmap-etc.yaml
index 1a20ea84..97bfe0f7 100644
--- a/cinder/templates/configmap-etc.yaml
+++ b/cinder/templates/configmap-etc.yaml
```

**Review by okozachenko1203 — DISMISSED** [review]

**Comment by mnaser** [comment]
recheck

**Review by mnaser — APPROVED** [review]

**Pull request merged**